app/stac_api/admin.py

```diff
@@ -113,8 +120,39 @@ class CollectionFilterForItems(AutocompleteFilter):
     field_name = 'collection'  # name of the foreign key
 
 
+# helper form to add an extra text_geometry field to ItemAdmin
+class ItemAdminForm(forms.ModelForm):
+    help_text = """Insert either:<br/>
+    - An extent in either WGS84 or LV95: "xmin, ymin, xmax, ymax"
+    where x is easting and y is northing<br/>
+    - A WKT polygon.
+    F.ex. "SRID=4326;POLYGON((5.96 45.82, 5.96 47.81, 10.49 47.81, 10.49 45.82, 5.96 45.82))"
+    <br/><br/><b>In any case the geometry will be saved as a WKT POLYGON in WGS84.</b>
+    """
+    text_geometry = CharField(
+        label='Geometry text',
+        required=False,
```

Comment:
I put False - the reason are the unittests of test_admin_page. The required is done by the validator

Reply:
Ok but I would add a comment here about that.

Reply:
If it's just because of the existing unit test, I'd rather mark it required and adapt the unit test. (Or did I misunderstand sth?)

Reply:
No, you are right - this is a workaround - the unittest should be able to treat the form (with extra field) as well. But I just wasn't able to do so...

Reply:
`======================================================================
ERROR: test_add_remove_item (tests.test_admin_page.AdminItemTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ltrea/.local/share/virtualenvs/service-stac-a9RlioXy/lib/python3.7/site-packages/moto/core/models.py", line 106, in wrapper
    result = func(*args, **kwargs)
  File "/home/ltrea/service-stac/app/tests/utils.py", line 97, in wrapper
    test_function(*args, **kwargs)
  File "/home/ltrea/service-stac/app/tests/test_admin_page.py", line 593, in test_add_remove_item
    item, data, link = self._create_item(self.collection, with_link=True)
  File "/home/ltrea/service-stac/app/tests/test_admin_page.py", line 171, in _create_item
    self.assertEqual(getattr(item, key), value, msg=f"Item field {key} value missmatch")
AttributeError: 'Item' object has no attribute 'text_geometry'
`

Reply:
If you set `require=True` then you need to do the following modification in unittest in `test_admin_page.py:_create_item`

```python
def _create_item(self, collection, with_link=False, extra=None):

        # Post data to create a new item
        # Note: the *-*_FORMS fields are necessary management form fields
        # originating from the AdminInline and must be present
        data = {
            "collection": collection.id,
            "name": "test_item",
            "geometry":
                "SRID=4326;POLYGON((5.96 45.82, 5.96 47.81, 10.49 47.81, 10.49 45.82, 5.96 45.82))",
            "text_geometry":
                "SRID=4326;POLYGON((5.96 45.82, 5.96 47.81, 10.49 47.81, 10.49 45.82, 5.96 45.82))",
            "properties_datetime_0": "2020-12-01",
            "properties_datetime_1": "13:15:39",
            "properties_title": "test",
            "links-TOTAL_FORMS": "0",
            "links-INITIAL_FORMS": "0",
        }
        if with_link:
            data.update({
                "links-TOTAL_FORMS": "1",
                "links-INITIAL_FORMS": "0",
                "links-0-href": "http://www.example.com",
                "links-0-rel": "example",
                "links-0-link_type": "example",
                "links-0-title": "Example test",
            })
        if extra:
            data.update(extra)
        response = self.client.post(reverse('admin:stac_api_item_add'), data)

        # Status code for successful creation is 302, since in the admin UI
        # you're redirected to the list view after successful creation
        self.assertEqual(response.status_code, 302, msg="Admin page failed to add new item")
        self.assertTrue(
            Item.objects.filter(collection=collection, name=data["name"]).exists(),
            msg="Admin page item added not found in DB"
        )
        item = Item.objects.get(collection=collection, name=data["name"])
        link = None

        if with_link:
            self.assertTrue(
                ItemLink.objects.filter(item=item, rel=data["links-0-rel"]).exists(),
                msg="Admin page Link added not found in DB"
            )
            link = ItemLink.objects.get(item=item, rel=data["links-0-rel"])

        # Check the item values
        for key, value in data.items():
            if key in [
                'collection',
                'id',
                'properties_datetime_0',
                'properties_datetime_1',
                'text_geometry'
            ]:
                continue
            if key.startswith('links-0-'):
                self.assertEqual(
                    getattr(link, key[8:]), value, msg=f"Item link field {key} value missmatch"
                )
            elif key.startswith('links-'):
                continue
            else:
                self.assertEqual(getattr(item, key), value, msg=f"Item field {key} value missmatch")

        return item, data, link
```
And it works ;-)

Reply:
Thanks - I should have seen this